apis/hive/v1/clusterdeployment_types.go

```diff
@@ -249,6 +249,12 @@ type ClusterPoolReference struct {
 	Namespace string `json:"namespace"`
 	// PoolName is the name of the ClusterPool for which the cluster was created.
 	PoolName string `json:"poolName"`
+	// PoolVersion is an opaque value representing the state of the important (to
+	// ClusterDeployments) fields of the ClusterPool at the time this CD was created. Comparing
+	// this value to the ClusterPool.Status.PoolVersion is a quick indication as to whether the
+	// ClusterDeployment's configuration matches that of the pool.
+	// +optional
+	PoolVersion string `json:"poolVersion,omitempty"`
```

Comment:
personally it feels odd to have an internal detail like that from the clusterpool to live here. i would have preferred an annotation instead of including that in object's spec api.

Reply:
I have a healthy dislike of annotations as a rule, but let me think about this some more.

Reply:
Thought about it some more. Totally agree. Changed it out.